### ISSUE TITLE
Basic Auth with traefik middleware

### ISSUE THREAD

**Issue by mcdermottpa** [swarmpit/swarmpit#594]
---------------------------------------------------
BUG REPORT
---------------------------------------------------

**Description**

I am attempting to add basic authentication using traefik's middleware. 

**Steps to reproduce the issue:**
1. add label: - traefik.http.routers.swarmpit-https.middlewares=admin-auth

**What happens:**

A basic auth dialog displays as expected, but once authenticated swarmpit login page is skipped, and the dashboard is displayed with no data populated. A red error box displays at the bottom of the page with the message "{}"

**What should happen:**

After http basic authentication swarmpit should operate normally.

**Additional information (e.g. docker version, cluster setup,...):**

Docker: 20.10.5, build 55c4c88
traefik: 2.2.11

**Comment by dcniko** [comment]
The way that Swarmpit works is by using ajax calls to receive data asynchronusly. When you enable basic-auth middleware those calls doesn't know that they need to authenticate. And simply they receive 401 error. What you need to do is to correclty set up traefik routers:

```
      labels:
        - traefik.enable=true

        - traefik.http.routers.swarmpit-ajax.rule=Host(`your.url.tld`) && Headers(`referer`, `https://your.url.tld/`)
        - traefik.http.routers.swarmpit-ajax.entrypoints=websecure
        - traefik.http.routers.swarmpit-ajax.tls=true
        - traefik.http.routers.swarmpit-ajax.tls.certresolver=letsencrypt

        - traefik.http.routers.swarmpit.rule=Host(`your.url.tld`)
        - traefik.http.routers.swarmpit.entrypoints=websecure
        - traefik.http.routers.swarmpit.tls=true
        - traefik.http.routers.swarmpit.tls.certresolver=letsencrypt
        - traefik.http.routers.swarmpit.middlewares=traefik-auth

        - traefik.http.services.swarmpit.loadbalancer.server.port=8080
```

I'm using labels but the same can be achived by TOML or YAML. You can notice that middleware is ommited whe ajax calls are expected. You can fiddle more with the rules (ie. use other header or some regexp) but that should get you going. This example should be included into Traefik documentation as it is not strightforward. I hope it will help you or others with similar issue.

**Comment by aert** [comment]
My 2 cents for new comers (like me), swarmpit has it's own authentication system, so no need for basic auth middleware.
Just set the admin username & pwd at first launch (the web UI will show the user creation form on first launch), then you're set.

**Comment by krystofbe** [comment]
Hi @dcniko, and thank you for your input. However, there is a significant security concern associated with the proposed solution.

The use of the "Referer" header for authentication or authorization is dangerous because it can be easily manipulated. Attackers can fake or spoof the "Referer" header to bypass security measures, making this method unreliable and insecure.

In the provided Traefik configuration:
```
- traefik.http.routers.swarmpit-ajax.rule=Host(`your.url.tld`) && Headers(`referer`, `https://your.url.tld/`)
```

The security is relying on the "Referer" header to allow or deny access to specific routes. If an attacker can change the "Referer" header to match the expected value, they can potentially gain unauthorized access to sensitive parts of the application. 

As @aert pointed out, swarmpit already has its own auth system and the provided solution does not really add any value here.
